packages/crud/src/vaadin-crud-dialog.js

```diff
@@ -41,15 +43,57 @@ const editorTemplate = html`
   </div>
 `;
 
+class CrudDialogOverlay extends DialogOverlay {
+  static get is() {
+    return 'vaadin-crud-dialog-overlay';
+  }
+
+  static get template() {
+    if (!memoizedTemplate) {
+      memoizedTemplate = super.template.cloneNode(true);
+      const contentPart = memoizedTemplate.content.querySelector('[part="content"]');
+      const defaultSlot = contentPart.querySelector('slot:not([name])');
+      contentPart.removeChild(defaultSlot);
+      contentPart.appendChild(editorTemplate.content.cloneNode(true));
+    }
+    return memoizedTemplate;
+  }
+}
+
+customElements.define('vaadin-crud-dialog-overlay', CrudDialogOverlay);
+
 class CrudDialog extends Dialog {
+  /**
+   * Override template to provide custom overlay tag name.
+   */
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: none;
+        }
+      </style>
+
+      <vaadin-crud-dialog-overlay
+        id="overlay"
+        on-opened-changed="_onOverlayOpened"
+        on-mousedown="_bringOverlayToFront"
+        on-touchstart="_bringOverlayToFront"
+        theme$="[[theme]]"
+        modeless="[[modeless]]"
+        with-backdrop="[[!modeless]]"
+        resizable$="[[resizable]]"
+        focus-trap
+      ></vaadin-crud-dialog-overlay>
+    `;
+  }
+
   ready() {
     super.ready();
 
-    this.renderer = (root) => {
-      if (!root.__dialogInitialized) {
-        root.__dialogInitialized = true;
-        root.$.content.appendChild(editorTemplate.content.cloneNode(true));
-      }
+    this.renderer = (_root) => {
```

Comment:
Didn't notice any difference with this removed and all the tests passed also

Reply:
Good catch, it's not needed indeed. Removed.